### PR TITLE
Improve cmd_help.yaml and fix random number generator

### PR DESCRIPTION
**Pull request by brightprogrammer**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
Addresses #3440 to improve `cmd_help.yaml` commands help. Also fixes a bug where random number generator command `%r` generated wrong output. Since there was not test to check the output, it slipped but now it's fixed.

Added support for 64bit random numbers. Current implementation takes OR of two 32 bit random numbers to generate a 64 bit random number.

**Closing issues**

#3440

### CODE REVIEW

**Comment by wargio** [comment]
what is the difference in the output?

File: librz/core/cmd/cmd_math.c [hunk 1]
```diff
@@ -165,25 +165,14 @@ RZ_IPI RzCmdStatus rz_set_active_tab_next_handler(RzCore *core, int argc, const
...
-		if (argc == 3) {
-			const char *p = argv[2];
-			b = (ut32)rz_num_math(core->num, out);
-			r = (ut32)rz_num_math(core->num, p) - b;
-		} else {
-			r = (ut32)rz_num_math(core->num, out);
-		}
-	}
+	// TODO: Add support for 64 bit random numbers
+	const char *lowlimit = argv[1];
```
Comment (XVilka) [inline librz/core/cmd/cmd_math.c:168]: Please add support for those 64bit numbers while you are on it.

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by brightprogrammer** [comment]
> what is the difference in the output?

For random numbers? Previously it didn't care about low and high values given as arguments. Now it is always in range.

File: librz/util/unum.c [hunk 2]
```diff
@@ -32,19 +31,25 @@ static void rz_srand(int seed) {
 #endif
 }
 
-static int rz_rand(int mod) {
+static ut64 rz_rand(ut64 mod) {
+	ut64 r;
 #if HAVE_ARC4RANDOM_UNIFORM
-	return (int)arc4random_uniform(mod);
+	r = (ut64)arc4random_uniform(mod);
```
Comment (XVilka) [inline librz/util/unum.c]: See also https://stackoverflow.com/questions/26549830/swift-random-number-for-64-bit-integers

File: librz/util/unum.c [hunk 3]
```diff
@@ -32,19 +31,25 @@ static void rz_srand(int seed) {
...
+	return r;
 }
 
 RZ_API void rz_num_irand(void) {
-	rz_srand(rz_time_now());
+	rz_num_srand(rz_time_now());
 }
 
-RZ_API int rz_num_rand(int max) {
+RZ_API ut64 rz_num_rand(ut64 max) {
```
Comment (XVilka) [inline librz/util/unum.c]: I think, it should be `ut32` for old function, also rename it to `rz_num_rand32()` and add a new one - `rz_num_rand64()`. Better to be explicit about the numbers size

File: test/unit/test_contrbtree.c [hunk 4]
```diff
@@ -25,15 +25,15 @@ bool test_rz_rbtree_cont_insert() {
 	RContRBTree *tree = rz_rbtree_cont_new();
 	ut32 i;
 	for (i = 0; i < 2000; i++) {
-		ut32 v = (ut32)rz_num_rand(UT32_MAX >> 1);
-		rz_rbtree_cont_insert(tree, (void *)(size_t)v, simple_cmp, NULL);
+		ut64 v = (ut64)rz_num_rand(UT64_MAX >> 1);
```
Comment (XVilka) [inline test/unit/test_contrbtree.c]: Then it will probably needs a new test for 64bit, instead of modifying old one

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by ret2libc — APPROVED** [review]

File: test/unit/test_contrbtree.c (hunk 4 above)
Comment (brightprogrammer) [reply test/unit/test_contrbtree.c]: this isn't a test for random numbers if I understand it correctly. This is to test RBTree's insert implementation. Do we really need to add another insert check with 64bit random numbers?

Comment (XVilka) [reply test/unit/test_contrbtree.c]: What I meant - keep the 32 bit implementation here, the `rz_num_rand32()`

File: librz/util/unum.c [hunk 5]
```diff
@@ -32,19 +31,43 @@ static void rz_srand(int seed) {
...
 RZ_API void rz_num_irand(void) {
-	rz_srand(rz_time_now());
+	rz_num_srand(rz_time_now());
+}
+
+// NOTE: The random generator will be seeded twice
+// but I don't think that'll be a problem since it'll
+// be seeded twice at max
+
+RZ_API ut32 rz_num_rand32(ut32 max) {
```
Comment (XVilka) [inline librz/util/unum.c:67]: Add brief Doxygen for both

File: test/unit/test_contrbtree.c [hunk 6]
```diff
@@ -25,15 +25,15 @@ bool test_rz_rbtree_cont_insert() {
 	RContRBTree *tree = rz_rbtree_cont_new();
 	ut32 i;
 	for (i = 0; i < 2000; i++) {
-		ut32 v = (ut32)rz_num_rand(UT32_MAX >> 1);
-		rz_rbtree_cont_insert(tree, (void *)(size_t)v, simple_cmp, NULL);
+		ut64 v = (ut64)rz_num_rand64(UT64_MAX >> 1);
```
Comment (XVilka) [inline test/unit/test_contrbtree.c]: Please use 32bit version here

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/util/unum.c [hunk 7]
```diff
@@ -435,6 +500,9 @@ static ut64 rz_num_op(RzNum *num, char op, ut64 a, ut64 b) {
 	return b;
 }
 
+/**
```
Comment (XVilka) [inline librz/util/unum.c]: Remove

**Review by XVilka — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**